tests/CommonExtractor.php

```diff
@@ -0,0 +1,497 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractorCommon\Tests;
+
+use Keboola\Component\UserException;
+use Keboola\Csv\CsvWriter;
+use Keboola\Csv\Exception as CsvException;
+use Keboola\DbExtractorCommon\Configuration\BaseExtractorConfig;
+use Keboola\DbExtractorCommon\Configuration\DatabaseParameters;
+use Keboola\DbExtractorCommon\Configuration\TableDetailParameters;
+use Keboola\DbExtractorCommon\Configuration\TableParameters;
+use Keboola\DbExtractorCommon\Exception\ApplicationException;
+use Keboola\DbExtractorCommon\Exception\DeadConnectionException;
+use Keboola\DbExtractorCommon\BaseExtractor;
+use Keboola\DbExtractorCommon\RetryProxy;
+
+class CommonExtractor extends BaseExtractor
+{
+    public const TYPE_AUTO_INCREMENT = 'autoIncrement';
+    public const TYPE_TIMESTAMP = 'timestamp';
+
+    /** @var \PDO|null */
+    private $connection;
+
+    /** @var array */
+    private $incrementalFetching;
+
+    public function extract(BaseExtractorConfig $config): array
+    {
+        $imported = [];
+        $outputState = [];
+        if ($config->isConfigRow()) {
+            $tableParameters = TableParameters::fromRaw($config->getParameters());
+            $exportResults = $this->extractTable($tableParameters);
+            if (isset($exportResults['state'])) {
+                $outputState = $exportResults['state'];
+                unset($exportResults['state']);
+            }
+            $imported = $exportResults;
+        } else {
+            foreach ($config->getEnabledTables() as $table) {
+                $exportResults = $this->extractTable($table);
+                $imported[] = $exportResults;
+            }
+        }
+
+        return [
+            'status' => 'success',
+            'imported' => $imported,
+            'state' => $outputState,
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getTables(array $tables = []): array
```

Comment:
Tohle je fakt mordor - vůbec není jasné, co všechno by ten extractor měl vracet... stačí string[]?

Reply:
A co když moje DB nepodporuje ROW COUNT? Vadí to? Jak to poznám?

Reply:
Toto definuje abstraktni metodu, takze na sovu db, at uz podporuje cokoli si to musis napsat sam.

To jaky to ma tvar se mi take moc nelibi a souvisi to s tim, ze nemame nijak definovane odpovedi alespon pro tyto sync akce. Jinak bych byl take za value objekty.

Reply:
imho to klidne muze vracet value objekt, akoratze bude dost slozitej, ale v principu vsechny db ex maji mit stejnou odpoved (nebo aspon stejny zaklad) na get tables

Reply:
Čekal bych něco jako `Tables[]`, přičemž existuje $table->getPlatformSpecificParams(), kde může být potenciálně cokoli.

Reply:
Jo, to mam dokumentovat.  napisu issue.

Reply:
mate pravdu ze ma kazdy implementace vratit stejna struktura, tak souhlasim ze nejlepsi je delat 2 value objects.  `TableMeta` a `ColumnMeta`.

Reply:
https://github.com/keboola/db-extractor-common/issues/100